packages/mui-joy/src/styles/components.d.ts

```diff
@@ -0,0 +1,18 @@
+import { CSSInterpolation } from '@mui/system';
+import { ButtonProps } from '../Button/ButtonProps';
+
+export type OverridesStyleRules<ClassKey extends string = string> = Record<
+  ClassKey,
+  CSSInterpolation
+>;
+
+export interface Components {
+  MuiButton?: {
+    defaultProps?: Partial<ButtonProps>;
+    styleOverrides?: Partial<OverridesStyleRules<''>>;
```

Comment:
Could we not add `styleOverrides` :)? Or if we add it, let's please restrict it to the `ClassKeys` without the state classes (`disabled`) for exmaple.

Reply:
Fixed!